# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 63670a1a8495
# Full Hash: 63670a1a8495ac749739992ce466dba1808260fb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-03-06 09:47:39
# Description:
#   Bug 1618325 - Check IsContextLost in ReadPixels. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65435
# ==============================================================================

diff -r 3c67e9f15e93 -r 63670a1a8495 dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Thu Mar 05 18:23:38 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Thu Mar 05 15:54:13 2020 +0000
@@ -3980,6 +3980,8 @@
 
 bool ClientWebGLContext::ReadPixels_SharedPrecheck(
     CallerType aCallerType, ErrorResult& out_error) const {
+  if (IsContextLost()) return false;
+
   if (mCanvasElement && mCanvasElement->IsWriteOnly() &&
       aCallerType != CallerType::System) {
     JsWarning("readPixels: Not allowed");
